libwild/src/output_section_id.rs

```diff
@@ -375,6 +375,7 @@ pub(crate) struct BuiltInSectionDetails {
     pub(crate) link: &'static [OutputSectionId],
     pub(crate) start_symbol_name: Option<&'static str>,
     pub(crate) end_symbol_name: Option<&'static str>,
+    pub(crate) synthetic_symbol_names: Option<&'static [&'static str]>,
```

Comment:
Is there intended to be a difference between `None` and `Some(&[])` here? If not, then we could get rid of the `Option`.

Is this intended to be different to `start_symbol_name`? i.e. is this supposed to be start-symbols that have provide-semantics? I'm unsure about the naming, since these symbols seem just as synthetic as the start/end symbols. I guess until such time as we support provide-semantics, one option would be to change `start_symbol_name: `Option<&'static str>` to `start_symbol_names: &'static [&'static str]`. Then, once we support provide-semantics, split it out into two separate fields. e.g. `start_symbol_names` and `provide_start_symbol_names` or something like that.